buffer/src/main/java/io/netty/buffer/Unpooled.java

```diff
@@ -262,38 +262,86 @@ public static ByteBuf wrappedBuffer(ByteBuffer... buffers) {
         return wrappedBuffer(buffers.length, buffers);
     }
 
+    // used to avoid temporary array allocation/copy when wrapping byte[][] and ByteBuffer[]
+    abstract static class WrappingIterable<T> implements Iterable<ByteBuf> {
+        private final T[] array;
+        private final int start;
+
+        protected WrappingIterable(T[] array) {
+            this.array = array;
```

Comment:
Is it really safe to not call `clone()` ?

Reply:
@normanmaurer I think so - `CompositeByteBuf` doesn't retain a reference to any `ByteBuf` array or `Iterable` passed to its constructors, it just extracts the contained `ByteBuf`s within the context of the ctor. Unless I am missing something (very possible!)

I've now addressed all the other comments.

Reply:
I think if this is the assumption (did not check this yet) we should add a comment. Also I think we should maybe better the Iterable into the CompositeByteBuf then as it has a clear contract with it.

Reply:
Agree with both... and actually I think maybe simpler would just be a generic package-private constructor for CBB which takes a `T[]` where `T extends ByteHolder` where the byte holder is internal interface with the wrap methods, will update this PR when I get a chance.